File: frontend/src/apiCalls/DTO/audioDTO.ts
// audioDTO.ts

export namespace AudioDTO {

  // DTO for Create Audio Request
  export interface CreateAudioRequest {
    userId: number;
    audioBase64String?: string;
    title?: string;
    album?: string;
  }

  // DTO for Create Audio Result
  export interface CreateAudioResult {
    audioId: number;
    statusCode: number;
    message?: string;
  }

  // DTO for Delete Audio Request
  export interface DeleteAudioRequest {
    audioId: number;
    userId: number;
  }

  // DTO for Delete Audio Result
  export interface DeleteAudioResult {
    statusCode: number;
    message?: string;
  }

  // DTO for Edit Audio Request
  export interface EditAudioRequest {
    audioId: number;
    userId: number;
    title?: string;
    audioBase64String?: string;
    album?: string;
  }

  // DTO for Edit Audio Result
  export interface EditAudioResult {
    statusCode: number;
    message?: string;
  }

  // DTO for Get Audio Request
  export interface GetAudioRequest {
    audioId: number;
  }

  // DTO for Get Audio Result
  export interface GetAudioResult {
    audioBase64String?: string;
    title?: string;
    album?: string;
    creator?: string;
    statusCode: number;
    message?: string;
  }
}

File: backend/DTO/AudioDTO.cs
namespace backend.DTO
{
  public class AudioRequest
  {
    public int? UserId { get; set; }
    public int? AudioId { get; set; }
    public string? AudioBase64String { get; set; }
    public string? Title { get; set; }
    public string? Album { get; set; }
  }

  public class AudioResponse
  {
    public AudioData? Audio { get; set; }
    public int StatusCode { get; set; }
    public string? Message { get; set; }
  }

  public class AudioData
  {
    public int AudioId { get; set; }
    public string? AudioBase64String { get; set; }
    public string? Title { get; set; }
    public string? Album { get; set; }
  }
}